python/tk_multi_publish2/processing/plugin_manager.py

```diff
@@ -85,11 +86,28 @@ def top_level_items(self):
         """
         return self._root_item.children
 
-    def remove_top_level_item(self, item):
+    def remove_top_level_item(self, item_to_remove):
         """
         Removed the given item from the list of top level items
         """
-        self._root_item.remove_item(item)
+
+        # since we keep a look of items created from browsed/dropped external
+        # files, we can check to see if the item to be deleted is one of those.
+        # if it is, we need to remove that item from the lookup and remove the
+        # path from the list of external files we're tracking. first, see if the
+        # item matches any items created for a path. there should only be one,
+        # but we'll check all items in the lookup to be sure.
+        paths_to_delete = []
+        for (path, item_for_path) in self._path_to_item_lookup.iteritems():
```

Comment:
could a path ever generate more than one item? i guess not.

Reply:
I spoke about this with Josh as well, I guess it could if the user designed the collector too?
For example, maybe the user might wish to create a second item when an image file is supplied, and it would create and publish an info/meta data file to sit alongside it, although I guess technically it would be a new path, it would be generated of the original?